src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1049,12 +1049,21 @@ namespace winrt::TerminalApp::implementation
         auto profileName = profile.Name();
         profileMenuItem.Text(profileName);
 
-        // If there's an icon set for this profile, set it as the icon for
-        // this flyout item
-        const auto& iconPath = profile.EvaluatedIcon();
-        if (!iconPath.empty())
+        // If a custom icon path has been specified, set it as the icon for
+        // this flyout item. Otherwise, if an icon is set for this profile, set that icon
+        // for this flyout item.
+        if (iconPathOverride.empty())
```

Comment:
My gut feeling would be that menu item icons are rare, so I'd restructure it like below and as a bonus eliminate the common statements (personal preference! you're free to disagree, and I wouldn't block over it)

```c++
auto iconPath = profile.EvaluatedIcon();
if (!iconPathOverride.empty()) [[unlikely]]
{
    iconPath = iconPathOverride;
}

if (!iconPath.empty)
{
    const auto icon = _CreateNewTabFlyoutIcon(iconPath);
    profileMenuItem.Icon(icon);
}
```

I know it wastes some cycles evaluating every icon that is ever _overridden_...

so maybe,

```c++
auto iconPath = iconPathOverride.empty() ? profile.EvaluatedIcon() : iconPathOverride;

if (!iconPath.empty())
{
    const auto icon = _CreateNewTabFlyoutIcon(iconPath);
    profileMenuItem.Icon(icon);
}
```

I'm just noodling

Reply:
whatever we end up doing here, we should propagate

Reply:
yeah good point, I have updated both occurrences: dd41243ed174c3df35939a983008e806b4e48055